voluptuous/validators.py

```diff
@@ -102,7 +102,7 @@ class Coerce(object):
         ...   validate('foo')
     """
 
-    def __init__(self, type: type, msg: typing.Optional[str] = None) -> None:
+    def __init__(self, type: type | typing.Callable, msg: typing.Optional[str] = None) -> None:
```

Comment:
```suggestion
    def __init__(self, type: typing.Union[type, typing.Callable], msg: typing.Optional[str] = None) -> None:
```

`|` operator for typing was introduced in [python 3.10](https://docs.python.org/3/whatsnew/3.10.html) via [PEP-604](https://peps.python.org/pep-0604/) - previously a `typing.Union[foo_type, bar_type]` was used (and still can be used in 3.10+).